src/antsibull_core/utils/hashing.py

```diff
@@ -8,14 +8,32 @@
 from __future__ import annotations
 
 import hashlib
+import typing as t
+from collections.abc import Mapping
 
 import aiofiles
 
 from .. import app_context
 
 
+class _AlgorithmData(t.NamedTuple):
```

Comment:
A [frozen dataclass](https://docs.python.org/3/library/dataclasses.html#dataclasses.dataclass) might be better than a named tuple, but that's up to you.